usbsdmux/__main__.py

```diff
@@ -64,17 +64,38 @@ def main():
     ctl = UsbSdMux(args.sg)
     mode = args.mode.lower()
 
-    if mode == "off":
-        ctl.mode_disconnect()
-
-    elif mode in ("dut", "client"):
-        ctl.mode_DUT()
-
-    elif mode == "host":
-        ctl.mode_host()
-
-    elif mode == "get":
-        print(ctl.get_mode())
+    try:
+        if mode == "off":
+            ctl.mode_disconnect()
+
+        elif mode in ("dut", "client"):
+            ctl.mode_DUT()
+
+        elif mode == "host":
+            ctl.mode_host()
+
+        elif mode == "get":
+            print(ctl.get_mode())
+
+    except FileNotFoundError as fnfe:
+        print(fnfe, file=sys.stderr)
+        sys.exit(fnfe.errno)
+    except PermissionError as perr:
+        print(perr, file=sys.stderr)
+        sys.exit(perr.errno)
+    except OSError as ose:
+        if ose.errno == 25:
+            print(ose, file=sys.stderr)
+            print(
+                f"Does '{args.sg}' really point to an USB-SD-Mux?",
+                file=sys.stderr,
+            )
+            sys.exit(ose.errno)
+        else:
+            raise ose
+
+    except Exception as exe:
+        raise exe
```

Comment:
Why is this needed? It shouldn't do anything...

Reply:
You are right. No need to raise it manually.